packages/@okta/vuepress-theme-prose/components/OnThisPageItem.vue

```diff
@@ -17,31 +17,46 @@ export default {
   data() {
     return {
       imActive: false,
-      iHaveChildrenActive: false
+      iHaveChildrenActive: false,
+      linkHash: '',
     }
   },
   components: {
     OnThisPageItem: () => import('../components/OnThisPageItem.vue'),
   },
   mounted() {
     this.setActiveData();
+    this.calculateLinkHash();
   },
   watch: {
-    activeAnchor: function (val) {
-      this.setActiveData();
+    $route(to, from){
+     this.imActive = this.isLinkActive(this.link, to.hash)
+    },
+    activeAnchor(){
+      if(this.activeAnchor){
+        // this.$router.push(this.activeAnchor)
+        this.imActive = this.isLinkActive(this.link)
+      }
     }
   },
   methods: {
-    isActive: function( node ) {
-      if(this.activeAnchor === null) {
-        return false;
+    calculateLinkHash: function(link){
+      const linkPathDeconstructed = this.link.path.split('/');
+      this.linkHash = linkPathDeconstructed[linkPathDeconstructed.length-1];
+    },
+    isLinkActive: function( node, toLinkHash = null  ) {
+      if(this.activeAnchor && this.activeAnchor === this.linkHash){ 
+        return true 
+      }
+      if(toLinkHash && toLinkHash === this.linkHash) {
```

Comment:
what is the use case for this check? (expecting `this.activeAnchor` check to be sufficient for detecting current hash match)